### PR TITLE
feature/poi_click (clean branch)

### PR DESCRIPTION
**Pull request by hadiyahya11**
When you click on the magnifier icons, the corresponding activity is displayed.I added a linear layout that stores the content fragment. When we click on the magnifier icon, i set the visibility of the mapbox layout to gone and make the content's view visible I added a button so that you can go back to the map which does the inverse : sets the content's view to gone and make the mapbox activity visible again.
Capture d’écran 2021-05-14 à 10 51 04

### CODE REVIEW

File: app/src/main/java/com/pindex/main/ui/activities/MapboxActivity.kt [hunk 1]
```diff
@@ -336,4 +393,51 @@ class MapboxActivity : AppCompatActivity() {
...
+    private fun setUpExitButton(): ImageButton {
+        val button = ImageButton(applicationContext)
+        button.setBackgroundColor(
+            ContextCompat.getColor(
+                applicationContext,
+                R.color.white
+            )
+        )
+
+        button.background = resources.getDrawable(R.drawable.ic_baseline_close_36)
```
Comment (ToufiPF) [inline app/src/main/java/com/pindex/main/ui/activities/MapboxActivity.kt:430]: And this should be ContextCompat.getDrawable(this, R.drawable.ic_baseline_close_36)

File: app/src/main/java/com/pindex/main/ui/activities/MapboxActivity.kt [hunk 2]
```diff
@@ -336,4 +393,51 @@ class MapboxActivity : AppCompatActivity() {
...
+            Location.distanceBetween(
+                featurePoint.latitude(), featurePoint.longitude(),
+                p.geopoint!!.latitude, p.geopoint!!.longitude, result
+            )
+            if (result[0] < min) {
+                min = result[0]
+                closestPoint = p
+            }
+        }
+        return closestPoint
```
Comment (ToufiPF) [inline app/src/main/java/com/pindex/main/ui/activities/MapboxActivity.kt:417]: Seems buggy to me here, shouldn't min be outside the loop?
Also, check out LocationUtils, there's a method to get the distance between two locations/points

Comment (hadiyahya11) [reply app/src/main/java/com/pindex/main/ui/activities/MapboxActivity.kt]: You are totally right about the min value. It should be outside the loop and set to max value.
I saw LocationUtils it has getdistance method indeed but it does not provide any improvement from Location

**Review by amomo25 — APPROVED** [review]
Nice PR, the use of @VisibleForTesting is really smart

**Review by amomo25 — APPROVED** [review]

**Review by lucasrlt — APPROVED** [review]
Nice feature, gg !

**Pull request merged**